### PR TITLE
Integ tests nm mainloop

### PR DESCRIPTION
**Pull request by EdDev**
tests, nm.testlib: Introduce mainloop_run decorator

Start using the decorator in the nm.bond_test integration module.

### CODE REVIEW

**Comment by EdDev** [comment]
Dependent on #347

File: tests/integration/nm/testlib.py [hunk 1]
```diff
@@ -25,3 +26,11 @@ def mainloop():
     yield
     success = mloop.run(timeout=5)
     assert success, 'Mainloop error: ' + mloop.error
+
+
+def mainloop_run(func):
```
Comment (tyll) [inline tests/integration/nm/testlib.py]: On first thought, it should be `run_mainloop`

Comment (EdDev) [reply tests/integration/nm/testlib.py]: I think decorators are not supposed to be verbs.
They are usually used to add a property/behaviour to a function/method/class.

Comment (tyll) [reply tests/integration/nm/testlib.py]: I see.

File: tests/integration/nm/testlib.py [hunk 2]
```diff
@@ -25,3 +26,11 @@ def mainloop():
...
+    @functools.wraps(func)
+    def wrapper_mainloop(*args, **kwargs):
```
Comment (tyll) [inline tests/integration/nm/testlib.py]: nitpick/not a must: IMHO `wrap_mainloop` or `mainloop_wrapper` would be better.

Comment (EdDev) [reply tests/integration/nm/testlib.py]: Hmm.. This is the func that wraps the original func and which will be called every time the original is referenced.
Based on that, it should probably be a verb, like the one you suggested above.
Not sure exactly, I used the pattern usef here: https://realpython.com/primer-on-python-decorators/
Other references do not help much (https://stackoverflow.com/questions/10957409/python-naming-conventions-in-decorators).

**Pull request merged**